consumer/consumererror/permanent.go

```diff
@@ -13,6 +13,8 @@ type permanent struct {
 
 // NewPermanent wraps an error to indicate that it is a permanent error, i.e. an
 // error that will be always returned if its source receives the same inputs.
+//
+// Deprecated: [v0.110.0] use Error instead.
```

Comment:
I believe `NewPermanent` is no longer needed if we use `Error`.  I marked this deprecated in this PR, but we could do it in a followup PR.

If we agree to do it in this PR I'll update all the usages to fix the linter.